### ISSUE TITLE
rasa train emits a domain error without additional detail when using an incorrect field name in an utter

### ISSUE THREAD

**Issue by bastbnl** [RasaHQ/rasa#8625]
**Rasa version**: 
```
Rasa Version      :         2.5.1
Minimum Compatible Version: 2.2.0a1
Rasa SDK Version  :         2.5.0
Rasa X Version    :         None
Python Version    :         3.7.10
Operating System  :         Linux-4.19.0-16-amd64-x86_64-with-debian-10.9
Python Path       :         /home/bastb/bot/venv/bin/python3
```

**Rasa SDK version** (if used & relevant): 
```
(venv) bastb@i7-nuc:~/bot/bot_nld$ pip freeze | grep rasa-sdk
rasa-sdk==2.5.0
```

**Rasa X version** (if used & relevant):

**Python version**:
```
(venv) bastb@i7-nuc:~/bot/bot_nld$ python --version
Python 3.7.10
```

**Operating system** (windows, osx, ...):
`Linux i7-nuc 4.19.0-16-amd64 #1 SMP Debian 4.19.181-1 (2021-03-19) x86_64 GNU/Linux`

**Issue**:
I was working on `domain.yml` and used an incorrect key below an `utter_`, like this:
```
utter_request_cancelled:
- text: yadieyadie ya
- utter: different words, same message 

```

`rasa train` then reports an expected error, but with very little detail:
```
(venv) bastb@i7-nuc:~/bot/bot_nld$ rasa train
YamlValidationException: Failed to validate '/home/bastb/bot/bot_nld/domain.yml'. Please make sure the file is correct and all mandatory parameters are specified. Here are the errors found during validation

```

**Error (including full traceback)**:
```
YamlValidationException: Failed to validate '/home/bastb/bot/bot_nld/domain.yml'. Please make sure the file is correct and all mandatory parameters are specified. Here are the errors found during validation

```

**Command or request that led to error**:
```
rasa train
```

**Content of configuration file (config.yml)** (if relevant):
```yml

```

**Content of domain file (domain.yml)** (if relevant):
```yml

```

The source is easily found when running `rasa train -v`, but I think it would be nice if the error message itself contained some additional detail

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @jupyterjazz will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by Imod7** [comment]
Thank you @bastbnl for the detailed explanation of the issue. This issue is moved to the inbox of the @RasaHQ/enable-squad and it will be prioritised. In the meantime If you have any ideas on how to correct it, you can also open a PR and link it to this issue. The responses of the `domain.yml` are validated against this yaml [schema](https://github.com/RasaHQ/rasa/blob/main/rasa/shared/nlu/training_data/schemas/responses.yml) file and the validation does not show any errors because this [variable](https://github.com/RasaHQ/rasa/blob/f9d229ff9d8bf65d4b4673d4a7ddde437cbcbb36/rasa/shared/utils/validation.py#L176) is empty.

**Comment by bastbnl** [comment]
I gave it a try in PR #8698 after some really, really lite testing on my end using a domain.yml which was succesfully used for training

**Comment by ancalita** [comment]
Hi @bastbnl I think this should have been resolved for `rasa train` too with this [PR](https://github.com/RasaHQ/rasa/pull/8618) that got released in version [2.6.1](https://github.com/RasaHQ/rasa/releases/tag/2.6.1) - could you please try upgrading and testing if you see the same issue?

**Comment by bastbnl** [comment]
You're right, @ancalita. The issue has been resolved. Thanks!
